app/src/pages/inside/dashboardItemPage/widgetsGrid/widget/widgetHeader/widgetHeader.jsx

```diff
@@ -90,9 +92,15 @@ export class WidgetHeader extends Component {
       onEdit,
       customClass,
       isPrintMode,
+      isDashboardShared,
     } = this.props;
 
     const isOwner = data.owner === userId;
+    const isWidgetDeletable = canDeleteWidget(
+      userRole,
+      projectRole,
+      isOwner || (data.shared && !isDashboardShared),
```

Comment:
Should user have a possibility to remove another user's shared widget from his own *shared* dashboard?